File: cPP_java_py_js-DSA/oops-class/class-object/Class.js
// how to create a javascript class and use it method 
/*
    class keyword : it is used to declare a class
    constructor (): constructor keyword is used for declaring default constructor

*/

class Animal{
    // data members
    typeOfAnimal


    // default constructor 
    constructor(animalType){
        console.log("constructor called");
        this.typeOfAnimal  = animalType;
    }


}

// creating a object based on animal class 

let lio = new Animal ("lion");
console.log("printing object based on class Animal :",lio);

File: cPP_java_py_js-DSA/oops-class/class-object/Class.cpp
#include<bits/stdc++.h>
using namespace std  ;

class Animal{
    public: 
        string typeOfAnimal;
        int wt ; 
        int height;
        bool isCarnivore;


    // default constructor 
    Animal(string typeOfAnimal){
        this->typeOfAnimal = typeOfAnimal;

    }

    

    // available method 

    // print animal Details 
    void printAnimalDetials(){
        cout<<"Type of Animal"<<this->typeOfAnimal;
        cout<<"Weight :"<<this->wt<<"KG"<<endl;
        cout<<"Height :"<<this->height<<"Feet"<<endl;
        if(this->isCarnivore){
            cout<<"Carnivore animal"<<endl;
        }
        else{
            cout<<"No it is Herbivore animal"<<endl;
        }
    }

};
int main (){
    cout<<"OOPS concept in cpp";
    Animal lio ("Lion") ;

    return 0 ;
}